lab04_example/CMakeLists.txt

```diff
@@ -1,206 +1,36 @@
+# Author: Luke Dennis l.dennis@unsw.edu.au
+
 cmake_minimum_required(VERSION 3.0.2)
 project(lab04_example)
 
-## Compile as C++11, supported in ROS Kinetic and newer
-# add_compile_options(-std=c++11)
+add_compile_options(-std=c++17)
 
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
 find_package(catkin REQUIRED COMPONENTS
-  geometry_msgs
   roscpp
   std_msgs
+  geometry_msgs
+  genmsg
```

Comment:
Most lines in this file need further comments (unless added in later commits)

Reply:
These are auto generated and walked through in the PublisherSubscriber tutorial on ROS http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29#roscpp_tutorials.2FTutorials.2FWritingPublisherSubscriber.Building_your_nodes